### DISCUSSION TITLE
The 'show' command cannot be used in bash scripts

### DISCUSSION THREAD

**Discussion by yelen103** [YosysHQ/yosys#4294] [Q&A]
Hi,
![image](https://github.com/YosysHQ/yosys/assets/100701710/7cbb3b50-799a-4169-a5db-2e9515d827fc)
As shown in the figure above, the show command in the bash script will get an error, but when I go to yosys and execute it manually, it is normal.
![image](https://github.com/YosysHQ/yosys/assets/100701710/2e0bf62e-e036-42b0-b859-3f5c93b4495e)

**Comment by jix** [comment]
Each `yosys` call in your bash script starts a completely new Yosys instance starting with an empty design, discarding the current state when it exits. If you want to run multiple commands in a row they need to be part of the same `yosys` call. E.g. this should work:

```
yosys -p "
	read_verilog ~/yosys/compare/cnt.v
	proc
	show
	write_json ~/yosys/compare/cnt_ast_1.json
"
```

If you want to save the state across multiple yosys calls you can use `write_rtlil saved.il` at the end of one call and `read_rtlil saved.il` at the start of a following call to save and restore the current design to a file.
